# Biology Tools

Software libraries for drug discovery and computational biology.

## Deep Learning for Life Sciences

| Tool | Description | Link |
|------|-------------|------|
| **DeepChem** | Comprehensive ML library for drug discovery | [deepchem.io](https://deepchem.io/) |
| **DGL-LifeSci** | GNN library for life science applications | [github](https://github.com/awslabs/dgl-lifesci) |
| **Chemprop** | Message passing neural networks | [github](https://github.com/chemprop/chemprop) |
| **TorchDrug** | GNN-based drug discovery library | [torchdrug.ai](https://torchdrug.ai/) |
| **DeepPurpose** | Deep learning for drug-target interaction | [github](https://github.com/kexinhuang12345/DeepPurpose) |

---

## Protein Structure Prediction

| Tool | Description | Link |
|------|-------------|------|
| **AlphaFold3** | Advanced modeling with ligands, nucleotides, PTMs | [github](https://github.com/google-deepmind/alphafold3) |
| **AlphaFold2** | Protein structure prediction | [alphafold.ebi.ac.uk](https://alphafold.ebi.ac.uk/) |
| **ESM/ESMFold** | Protein language models & fast structure prediction | [github](https://github.com/facebookresearch/esm) |
| **ColabFold** | Community notebooks for AF2, ESMFold, RoseTTAFold | [github](https://github.com/sokrypton/ColabFold) |
| **OpenFold** | PyTorch reimplementation of AlphaFold2 | [github](https://github.com/aqlaboratory/openfold) |
| **RoseTTAFold** | Three-track neural network for structure | [github](https://github.com/RosettaCommons/RoseTTAFold) |
| **OmegaFold** | Single-sequence structure prediction | [github](https://github.com/HeliXonProtein/OmegaFold) |
| **Boltz-1** | Unified prediction for proteins, RNA, DNA, ligands | [github](https://github.com/jwohlwend/boltz) |
| **Chai-1** | Proteins, small molecules, DNA, RNA with restraints | [github](https://github.com/chaidiscovery/chai-lab) |

---

## Protein Design

| Tool | Description | Link |
|------|-------------|------|
| **ProteinMPNN** | Message passing for inverse folding | [github](https://github.com/dauparas/ProteinMPNN) |
| **RFDiffusion** | Diffusion model for backbone generation | [github](https://github.com/RosettaCommons/RFdiffusion) |
| **ColabDesign** | Notebooks for AfDesign, TrDesign, ProteinMPNN | [github](https://github.com/sokrypton/ColabDesign) |
| **ESM-IF1** | Inverse folding with ESM | [github](https://github.com/facebookresearch/esm#inverse-folding-) |
| **EvoDiff** | Discrete diffusion for sequences | [github](https://github.com/microsoft/evodiff) |
| **ProtGPT2** | GPT-2 based sequence generation | [huggingface](https://huggingface.co/nferruz/ProtGPT2) |
| **BindCraft** | Binder design | [github](https://github.com/martinpacesa/BindCraft) |
| **LM-Design / ByProt** | Language model design | [github](https://github.com/BytedProtein/ByProt) |
| **ProteinSolver** | Graph neural network for constraint-based design | [github](https://github.com/ostrokach/proteinsolver) |
| **ECNet** | Fine-tunable fitness/function prediction | [github](https://github.com/luoyunan/ECNet) |

---

## Protein Language Models

| Model | Description | Link |
|-------|-------------|------|
| **ESM-2** | Protein embeddings (8M-15B params) | [github](https://github.com/facebookresearch/esm) |
| **ProtTrans** | Transformer embeddings for proteins | [github](https://github.com/agemagician/ProtTrans) |
| **AMPLIFY** | ESM2 reimplementation with open training | [github](https://github.com/chandar-lab/AMPLIFY) |
| **PoET** | Variant effect prediction and generation | [github](https://github.com/OpenProteinAI/PoET) |
| **EvoProtGrad** | MCMC-based directed evolution | [github](https://github.com/NREL/EvoProtGrad) |

---

## Multimer & Complex Prediction

| Tool | Description | Link |
|------|-------------|------|
| **AlphaFold2-Multimer** | Protein complex prediction | [github](https://github.com/deepmind/alphafold) |
| **AlphaPulldown** | Protein-protein interaction screening | [github](https://github.com/KosinskiLab/AlphaPulldown) |
| **AF2Complex** | Complex assembly | [github](https://github.com/FreshAirTonight/af2complex) |
| **PeSTo** | Interface residue prediction | [github](https://github.com/LBM-EPFL/PeSTo) |

---

## Visualization & Analysis

| Tool | Description | Link |
|------|-------------|------|
| **PyMOL** | Molecular visualization | [pymol.org](https://pymol.org/) |
| **ChimeraX** | Molecular visualization for structural biology | [rbvi.ucsf.edu](https://www.rbvi.ucsf.edu/chimerax/) |
| **BioPython** | Python tools for biological computation | [biopython.org](https://biopython.org/) |
| **ProDy** | Protein dynamics analysis | [prody.csb.pitt.edu](http://prody.csb.pitt.edu/) |

---

## Single-Cell Analysis

| Tool | Description | Link |
|------|-------------|------|
| **Scanpy** | Single-cell analysis in Python | [scanpy.readthedocs.io](https://scanpy.readthedocs.io/en/stable/) |
| **Seurat** | Single-cell analysis in R | [satijalab.org/seurat](https://satijalab.org/seurat/) |
| **Squidpy** | Spatial transcriptomics analysis | [squidpy.readthedocs.io](https://squidpy.readthedocs.io/) |
| **scGPT** | Transformer model for single-cell | [github](https://github.com/bowang-lab/scGPT) |
| **scFoundation** | Single-cell foundation model | [github](https://github.com/biomap-research/scFoundation) |
| **scPRINT** | Cell imputation model | [github](https://github.com/cantinilab/scPRINT) |

---

## Drug Discovery Platforms

| Tool | Description | Link |
|------|-------------|------|
| **RDKit** | Core cheminformatics toolkit | [rdkit.org](https://www.rdkit.org/) |
| **Open Drug Discovery Toolkit** | ADMET prediction | [github](https://github.com/oddt/oddt) |
| **TDC** | Therapeutics Data Commons | [tdcommons.ai](https://tdcommons.ai/) |

---

## Drug-Target Interaction

| Tool | Description | Link |
|------|-------------|------|
| **NeoDTI** | Drug-target interaction prediction | [github](https://github.com/FangpingWan/NeoDTI) |
| **TransformerCPI** | Transformer for compound-protein interaction | [github](https://github.com/lifanchen-simm/transformerCPI) |
| **drGAT** | Attention-based drug response predictor | [github](https://github.com/inoue0426/drGAT) |

---

## Biological Language Models

| Model | Description | Link |
|-------|-------------|------|
| **ESM-2** | Protein embedding models | [github](https://github.com/facebookresearch/esm) |
| **ChemBERTa-2** | Chemical SMILES embeddings | [github](https://github.com/seyonechithrananda/bert-loves-chemistry) |
| **BioGPT** | Biomedical text generation | [github](https://github.com/microsoft/BioGPT) |
| **GeneGPT** | Biomedical information system | [github](https://github.com/ncbi/GeneGPT) |
| **GenePT** | Single-cell foundation model | [github](https://github.com/yiqunchen/GenePT) |

---

## Network & Pathway Analysis

| Tool | Description | Link |
|------|-------------|------|
| **STRING** | Protein-protein interaction networks | [string-db.org](https://string-db.org/) |
| **Cytoscape** | Network visualization and analysis | [cytoscape.org](https://cytoscape.org/) |
| **PathwayCommons** | Biological pathway data | [pathwaycommons.org](https://www.pathwaycommons.org/) |

---

## Awesome Lists

| List | Focus |
|------|-------|
| [awesome-protein-design](https://github.com/johnnytam100/awesome-protein-design) | Protein design papers |
| [awesome-protein-design-software](https://github.com/pansapiens/awesome-protein-design-software) | Protein design tools |
| [awesome-AI-based-protein-design](https://github.com/opendilab/awesome-AI-based-protein-design) | AI protein design |
| [awesome-computational-biology](https://github.com/inoue0426/awesome-computational-biology) | Computational biology resources |
| [awesome-small-molecule-ml](https://github.com/benb111/awesome-small-molecule-ml) | Drug discovery ML |
| [awesome-drug-discovery](https://github.com/yboulaamane/awesome-drug-discovery) | Drug discovery tools |

---

## Getting Started

```bash
conda create -n bio python=3.10
conda activate bio

# Core tools
pip install deepchem rdkit

# For protein work
pip install biopython fair-esm

# For single-cell
pip install scanpy squidpy
```
